homeassistant/components/google_sheets/__init__.py

```diff
@@ -103,13 +99,9 @@ async def append_to_sheet(call: ServiceCall) -> None:
             call.data[DATA_CONFIG_ENTRY]
         )
         if not entry:
-            raise HomeAssistantError(
-                "Invalid config entry: {call.data[DATA_CONFIG_ENTRY]}"
-            )
+            raise ValueError("Invalid config entry: {call.data[DATA_CONFIG_ENTRY]}")
         if not (session := hass.data[DOMAIN].get(entry.entry_id)):
-            raise HomeAssistantError(
-                "Config entry not loaded: {call.data[DATA_CONFIG_ENTRY]}"
-            )
+            raise ValueError("Config entry not loaded: {call.data[DATA_CONFIG_ENTRY]}")
```

Comment:
This doesn't have to be user caused. It could be. I don't have a right answer for this now.

Reply:
I don't think this is likely to happen. If the config entry gets removed but then gets called by an automation, it would be caught with the previous check as an invalid config entry.